ui/app/AppLayouts/Profile/helpers/ContactDetails.qml

```diff
@@ -0,0 +1,113 @@
+import QtQuick 2.15
+
+import StatusQ 0.1
+import StatusQ.Core.Utils 0.1
+
+import AppLayouts.Profile.stores 1.0
+
+import utils 1.0
+
+QObject {
+    id: root
+
+    required property ContactsStore contactsStore
+    required property ProfileStore profileStore
+    required property string publicKey
+
+    readonly property alias loading: d.loading
+
+    // model properties
+    readonly property string displayName: d.contactDetails.displayName ?? ""
+    readonly property string ensName: d.contactDetails.ensName ?? ""
+    readonly property bool ensVerified: d.contactDetails.isEnsVerified ?? false
+    readonly property string localNickname: d.contactDetails.localNickname ?? ""
+    readonly property string alias: d.contactDetails.alias ?? ""
+    readonly property string icon: d.contactDetails.icon ?? ""
+    readonly property int colorId: d.contactDetails.colorId ?? 0
+    readonly property var colorHash: d.contactDetails.colorHash ?? []
+    readonly property int onlineStatus: d.contactDetails.onlineStatus ?? Constants.onlineStatus.inactive
+    readonly property bool isContact: d.contactDetails.isContact ?? false
+    readonly property bool isCurrentUser: d.contactDetails.isCurrentUser ?? false
+    readonly property bool isVerified: d.contactDetails.isVerified ?? false
+    readonly property bool isUntrustworthy: d.contactDetails.isUntrustworthy ?? false
+    readonly property bool isBlocked: d.contactDetails.isBlocked ?? false
+    readonly property int contactRequestState: d.contactDetails.contactRequest ?? Constants.ContactRequestState.None
+    readonly property int incomingVerificationStatus: d.contactDetails.incomingVerificationStatus ?? Constants.verificationStatus.unverified
+    readonly property int outgoingVerificationStatus: d.contactDetails.outgoingVerificationStatus ?? Constants.verificationStatus.unverified
+    readonly property string defaultDisplayName: d.contactDetails.defaultDisplayName ?? ""
+    readonly property string optionalName: d.contactDetails.optionalName ?? ""
+    readonly property int lastUpdated: d.contactDetails.lastUpdated ?? 0
+    readonly property int lastUpdatedLocally: d.contactDetails.lastUpdatedLocally ?? 0
+    readonly property string thumbnailImage: d.contactDetails.thumbnailImage ?? ""
+    readonly property string largeImage: d.contactDetails.largeImage ?? ""
+    readonly property bool isContactRequestReceived: d.contactDetails.isContactRequestReceived ?? false
+    readonly property bool isContactRequestSent: d.contactDetails.isContactRequestSent ?? false
+    readonly property bool isSyncing: d.contactDetails.isSyncing ?? false
+    readonly property bool removed: d.contactDetails.isRemoved ?? false
+    readonly property int trustStatus: d.contactDetails.trustStatus ?? Constants.trustStatus.unknown
+    readonly property string bio: d.contactDetails.bio ?? ""
+
```

Comment:
I'd also add the synthesized `displayName` here (`ProfileUtils.displayName()`) for convenience

Reply:
The `defaultDisplayName` already contains that logic.

Reply:
This is how it's computed
```
proc userExtractedName(contact: ContactsDto): string =
  if(contact.name.len > 0 and contact.ensVerified):
    result = contact.name
  elif contact.displayName.len > 0:
    result = contact.displayName
  else:
    result = contact.alias

proc userDefaultDisplayName*(contact: ContactsDto): string =
  if(contact.localNickname.len > 0):
    result = contact.localNickname
  else:
    result = userExtractedName(contact)
```